src/logic/function/basic.lean

```diff
@@ -357,6 +357,22 @@ rfl
 
 end bicomp
 
+section uncurry
+
+variables {α β γ δ : Type*}
+
+/-- Records a way to turn an element of `α` into a function from `β` to `γ`. -/
+class has_uncurry (α : Type*) (β : out_param Type*) (γ : out_param Type*) := (uncurry : α → (β → γ))
+
+notation `↿`:max x:max := has_uncurry.uncurry x
```

Comment:
Can this be a local(ized) notation? This seems like a pretty niche thing to do.

Reply:
I don't mind using localization but this id not niche at all, it can be used to replace all the contorsion we use to talk about multi-variable (uniformly) continuous or differentiable functions and probably also in algebra.